data/abilities.ts

```diff
@@ -2151,14 +2158,18 @@ export const Abilities: {[abilityid: string]: AbilityData} = {
 	},
 	multitype: {
 		// Multitype's type-changing itself is implemented in statuses.js
+		neverChanges: true,
 		name: "Multitype",
 		rating: 4,
 		num: 121,
 	},
 	mummy: {
 		name: "Mummy",
 		onDamagingHit(damage, target, source, move) {
-			if (move.flags['contact'] && source.ability !== 'mummy') {
+			if (source.getAbility().neverChanges && source.ability !== 'mummy') {
```

Comment:
```suggestion
			if (source.getAbility().neverChanges || source.ability === 'mummy') {
```
The old code filtered out Pokémon that had the ability Mummy. What you have now would filter out every Pokémon that _doesn’t_ have Mummy, I think.

Reply:
He's doing an early return for when the attacker's Ability can't be changed into Mummy.

Reply:
No, I think MacChaeger is right.

I could cache the `getAbility` though.

```suggestion
			const sourceAbility = source.getAbility();
			if (sourceAbility.neverChanges || sourceAbility.id === 'mummy') {
```

Reply:
Actually this would depend on: "If the attacker's Ability is Gastro Acid Mummy, will the defender's Mummy activate?" If yes, my code is correct.

Reply:
@MacChaeger's code is correct, but his comment was confusing.

Reply:
Zarel's correct, so I've gone and done it his way.